mlflow/dspy/callback.py

```diff
@@ -227,23 +226,20 @@ def on_evaluate_start(self, call_id: str, instance: Any, inputs: dict[str, Any])
         if not get_autologging_config(FLAVOR_NAME, "log_evals"):
             return
 
+        key = "eval"
+        if callback_metadata := inputs.get("callback_metadata"):
+            if "metric_key" in callback_metadata:
+                key = callback_metadata["metric_key"]
         if self.optimizer_stack_level > 0:
-            key = "eval"
-            if callback_metadata := inputs.get("callback_metadata"):
-                if "metric_key" in callback_metadata:
-                    key = callback_metadata["metric_key"]
             with _lock:
                 # we may want to include optimizer_stack_level in the key
                 # to handle nested optimization
                 step = self._evaluation_counter[key]
                 self._evaluation_counter[key] += 1
-            run = mlflow.start_run(run_name=f"{key}_{step}", nested=True)
             self._call_id_to_metric_key[call_id] = (key, step)
-            self._call_id_to_run_id[call_id] = run.info.run_id
+            mlflow.start_run(run_name=f"{key}_{step}", nested=True)
```

Comment:
Just for my understanding, there's always a parent run when on_evaluate_start is called?

Reply:
No, it's not always. But in my understanding there is no harm to create a root run with `nested=True`

Reply:
Interesting, seems yes `nested=True` doesn't do anything if there's no parent run